modules/photons_control/device_finder.py

```diff
@@ -541,13 +576,86 @@ def location_id(self):
             return sb.NotSpecified
         return self.location.uuid
 
+    @property
+    def product_id(self):
+        if self.product is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.pid
+
+    @property
+    def product_name(self):
+        if self.product_id is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.friendly
+
+    @property
+    def product_type(self):
+        if self.product is not sb.NotSpecified:
+            return (
+                ProductType.LIGHT.value
+                if Products[VendorRegistry.LIFX, self.product.pid].cap.is_light
+                else ProductType.SWITCH.value
+            )
+        else:
+            return sb.NotSpecified
+
+    @property
+    def firmware_version(self):
+        if self.firmware is sb.NotSpecified:
+            return sb.NotSpecified
+        return str(self.firmware)
+
+    @property
+    def cap(self):
```

Comment:
let's change, in a PR before this one this property to be like `abilities` so that we can replace it with a cap that is `self.product.cap(*self.firmware_version)` so it's an actual cap object

Reply:
call it abilities rather than cap so you can make a property called cap that is a cap object rather than a list of strings.

Reply:
Ah, gotcha. I thought there was another property called `abilities` somewhere in the desired format. :) The property `capabilties` is already a `cap` object. I only kept `cap` for backwards compatibility.

Reply:
the purpose of the list of strings remains but the word cap is "soft-reserved" as such for the cap object and it was a mistake for me to name this method cap. Hence a separate PR before this one to rename this property to abilities so that cap is not used for this PR and this PR can then create a cap property that returns a cap object

Reply:
Righto.